### DISCUSSION TITLE
Any plan to make the Lua session persistent?

### DISCUSSION THREAD

**Discussion by user202729** [Witiko/lt3luabridge#16] [Ideas]
As far as I can see, currently each call to Luabridge in non-LuaTeX engine invokes a separate write18, which invokes a separate Lua script.

So if you e.g. set "global" `x=1` in the first block, then read it in the second block it will be nil.

Is there any plan to support this case? (for example by starting a daemon process with `start lua ...` in Windows cmd or `lua ... &` in bash-like shells, then somehow get the PID of the TeX process and watch until it exits. A bit difficult to implement however)

**Comment by Witiko** [comment]
@user202729 Thank you for the feature request. At the moment, there is no such plan. The individual calls to the `\luabridge_now:n` functions are supposed to invoke a separate interpreter. Actually, it might be useful to enforce this by sandboxing the direct Lua calls (see https://github.com/Witiko/lt3luabridge/pull/14), so that they cannot set global variables.

**Comment by Witiko** [comment]
However, I would not mind separate functions (such as `\luabridge_persistent_now:n`) that would use a persistent Lua session if somebody would like to pick this up.
